x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.ts

```diff
@@ -303,6 +408,17 @@ export const RoleMappingsLogic = kea<MakeLogicType<RoleMappingsValues, RoleMappi
       const roleMapping = values.roleMappings.find(({ id }) => id === roleMappingId);
       if (roleMapping) actions.setRoleMapping(roleMapping);
     },
+    initializeSingleUserRoleMapping: ({ roleMappingId }) => {
+      const singleUserRoleMapping = values.singleUserRoleMappings.find(
+        ({ roleMapping }) => roleMapping.id === roleMappingId
+      );
```

Comment:
[not a change request] in a future refactor, do you think it would be worth storing a map copy of of user roles with `id` as key so lookup is immediate rather than having to search through an array?... 🤔 

... i'm definitely overthinking this. just give me my own paragraph in [this blog post](https://www.darkcoding.net/software/a-day-in-the-life-of-a-professional-software-engineer/) already :neckbeard: